### ISSUE TITLE
Error: RenderError("No vertices in MeshBuilder") when vertices are in wrong order

### ISSUE THREAD

**Issue by 00benallen** [ggez/ggez#628]
I'm building my first game with ggez, and found I needed to draw a trapezoid.

Here's the basic idea of what I was doing:

```
fn create_trapezoid_points(&self) -> [na::Point2<f32>; 4] {
   //blah, returns 4 points in an array
}

fn draw(&self, ctx: &mut Context) -> GameResult {
   let points = &self.create_trapezoid_points();

   let trapezoid_mesh = graphics::Mesh::new_polygon(
      ctx, 
      graphics::DrawMode::fill(), 
      &self.create_wall_points(), 
      [0.0, 0.0, 1.0, 1.0].into()
   )?;

   graphics::draw(ctx, &wall_mesh, (na::Point2::new(0.0, 0.0),))?;
   Ok(())
}
```

When running my game, I would get this error: 
`Error: RenderError("No vertices in MeshBuilder")` and the game would crash.

After debugging this for a while, I guessed that my points needed to be in clockwise order (I have experience with lower level graphics programming with openGL and the like). This fixed the problem, and I get my trapezoids now, but it was a little frustrating.

The requirement on the order of the points for `new_polygon` is not in the documentation for that function, AND the error message I'm getting is not very helpful.

How about `Error: RenderError("Vertices could not be drawn")` at least, or even better a more complete printout with the vertices in question and what's wrong with them?

**Comment by icefoxen** [comment]
Thank you!  We can definitely have better error messages and docs here.

**Comment by 00benallen** [comment]
I'd be willing to help btw, doesn't seem like a particularly arduous task

**Comment by icefoxen** [comment]
Sure, if you want.

**Comment by icefoxen** [comment]
Added some checks and better error messages, plus documentation.

However, related: #639 .  The underlying issue is that lyon is allowed to decide "there's no vertices to output here and that's technically okay".  For us though, it's usually not technically okay.

Added some more checks in `MeshBuilder::build()` for these cases, but they need unit tests.

**Comment by icefoxen** [comment]
Should now be resolved by 259e1aa .  Made error messages more helpful, docs more explicit, and error-checked what things I could.  Unfortunately CCW points isn't actually something I can check for easily, afaict?  But that constraint is at least documented now.  @00benallen Let me know if you have any suggestions for things I've missed.
